infrastructure/quick-deploy/aws/all/providers.tf

```diff
@@ -5,18 +5,30 @@ provider "aws" {
 
 provider "kubernetes" {
   host                   = module.eks.cluster_endpoint
-  cluster_ca_certificate = base64decode(module.eks.certificate_authority[0].data)
-  token                  = module.eks.token
-  insecure               = false
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
```

Comment:
Why did you need to change?

Reply:
I followed the official included exemple from the module
in v17 it was : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v17.24.0/examples/managed_node_groups/main.tf#L102
in v19 it is now : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v19.10.0/examples/self_managed_node_group/main.tf#L5